pkg/configuration/router.go

```diff
@@ -58,7 +97,158 @@ func updateOrganizationConfig(logger log.Logger, repo Repository) http.HandlerFu
+			moovhttp.Problem(w, fmt.Errorf("problem reading logo file - error=%v", err))
+			return
+		}
+
+		w.Header().Set("Content-Type", rdr.ContentType())
 		w.WriteHeader(http.StatusOK)
-		json.NewEncoder(w).Encode(cfg)
+		w.Write(fBytes)
 	}
 }
+
+func uploadOrganizationLogo(logger log.Logger, repo Repository, bucketFactory storage.BucketFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		requestID := moovhttp.GetRequestID(r)
+		organization := route.GetOrganization(w, r)
+		if organization == "" {
+			logger.Log("configuration", "upload organization logo called with no organization header", "requestID", requestID)
+			return
+		}
+
+		file, _, err := r.FormFile("file")
+		if file == nil || err != nil {
+			logger.Log("configuration", errMissingFile, "error", err, "organization", organization, "requestID", requestID)
+			moovhttp.Problem(w, errMissingFile)
+			return
+		}
+		defer file.Close()
+
+		// Detect the content type by reading the first 512 bytes of r.Body (read into file as we expect a multipart request)
+		buf := make([]byte, 512)
+		if _, err := file.Read(buf); err != nil && err != io.EOF {
+			logger.Log("configuration", "problem reading file", "error", err, "organization", organization, "requestID", requestID)
+			moovhttp.Problem(w, err)
+			return
+		}
+		contentType := http.DetectContentType(buf)
+
+		ext, supported := contentTypeFileExtensions[contentType]
+		if !supported {
+			logger.Log("configuration", "unsupported content type for logo image file", "error", err, "contentType", contentType, "organization", organization, "requestID", requestID)
+			moovhttp.Problem(w, errUnsupportedType)
+			return
+		}
+
+		originalConfig, err := repo.Get(organization)
+		if err != nil {
+			logger.Log("configuration", "error retrieving organization configuration", "error", err, "organization", organization, "requestID", requestID)
+			moovhttp.Problem(w, err)
+			return
+		}
+		wasUpdated := (originalConfig.LogoFile != "") // keep track so we can return appropriate HTTP status code
+		originalConfig.LogoFile = fmt.Sprintf("organization-%s-logo%s", organization, ext)
+
+		updatedCfg, err := repo.Update(organization, originalConfig)
+		if err != nil {
+			logger.Log("configuration", "problem updating configuration", "error", err, "organization", organization, "requestID", requestID)
+			moovhttp.Problem(w, err)
+			return
+		}
+
+		bucket, err := bucketFactory()
+		if err != nil {
+			logger.Log("configuration", "problem uploading logo image", "error", err, "organization", organization, "requestID", requestID)
+			moovhttp.Problem(w, err)
+			return
+		}
+		defer bucket.Close()
+
+		// Write our document from the request body
+		ctx, cancelFn := context.WithTimeout(r.Context(), 60*time.Second)
+		defer cancelFn()
+
+		writer, err := bucket.NewWriter(ctx, makeDocumentKey(organization, updatedCfg.LogoFile), &blob.WriterOptions{
```

Comment:
I was thinking we could store the organizationID in the path, then it'd look like. Sorry to only mention this after the code is written. 

```
gs://<bucket>/organizations/{organizationID}/logo
```

That removes the need to store anything in the DB and allows us to quickly scan that bucket if needed. I don't think there's a reason to store the revision history in the bucket.